crates/ty_python_semantic/src/program.rs

```diff
@@ -89,11 +104,48 @@ impl Program {
 #[derive(Clone, Debug, Eq, PartialEq)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize))]
 pub struct ProgramSettings {
-    pub python_version: PythonVersion,
+    pub python_version: PythonVersionWithSource,
     pub python_platform: PythonPlatform,
     pub search_paths: SearchPathSettings,
 }
 
+#[derive(Clone, Debug, Eq, PartialEq, Default)]
+pub enum ValueSource {
+    /// Value loaded from a project's configuration file.
+    File(Arc<SystemPathBuf>, Option<TextRange>),
+
+    /// The value comes from a CLI argument, while it's left open if specified using a short argument,
+    /// long argument (`--extra-paths`) or `--config key=value`.
+    Cli,
+
+    /// We fell back to a default value because the value was not specified via the CLI or a config file.
+    #[default]
+    Default,
+}
+
+#[cfg_attr(feature = "serde", derive(serde::Serialize))]
```

Comment:
It seems that `ProgramSetting's` `Serialize` and `Deserialize` implementation are never used. Can we remove them instead of making this type serializable